packages/hydrojudge/src/judge/generate.ts

```diff
@@ -0,0 +1,96 @@
+import { tmpdir } from 'os';
+import path from 'path';
+import fs from 'fs-extra';
+import { STATUS } from '@hydrooj/utils/lib/status';
+import { SystemError } from 'hydrooj';
+import { runQueued } from '../sandbox';
+import client from '../sandbox/client';
+import signals from '../signals';
+import { JudgeTask } from '../task';
+import { parseMemoryMB, parseTimeMS } from '../utils';
+
+export const judge = async (ctx: JudgeTask) => {
+    const tmp = path.join(tmpdir(), `${ctx.request.rid}.zip`);
+    ctx.stat.judge = new Date();
+    ctx.next({ status: STATUS.STATUS_COMPILING });
+    if (!('content' in ctx.code)) throw new SystemError('Unsupported input');
+    const [generator, std] = ctx.code.content.toString().split('\n').map((i) => i.trim());
+    if (generator.includes('/') || generator === '..') throw new SystemError('Invalid input');
+    if (std.includes('/') || std === '..') throw new SystemError('Invalid input');
+    const [executeGenerator, executeStd] = await Promise.all([
+        ctx.compileLocalFile('generator', generator),
+        ctx.compileLocalFile('std', std),
+    ]);
+    ctx.next({ status: STATUS.STATUS_JUDGING, progress: 0 });
+    const { address_space_limit, process_limit } = ctx.session.getLang(ctx.lang);
+    const content = `#!/bin/bash
+for i in {1..10}; do
+    ${executeGenerator.execute} $i >$i.in
+done
+for i in *.in; do
+    [ -f "$i" ] || break
+    ${executeStd.execute} <$i >\${i%.in}.out
+done
+zip data.zip *
+ls`;
+    const res = await runQueued(
```

Comment:
I don't really agree with generating and executing all test cases in single execution as it creates an all-or-none situation where it returns all test cases or none if there's any error. At the same time, the generating process cannot be parallelized to scale on multi-core CPUs. 

I would suggest to separate the generating process with the execution of the std program individually with respect of time & memory limit by problem config so that unsuccessful execution can be well tracked through ui.